src/v/cluster/topic_table_probe.cc

```diff
@@ -9,14 +9,133 @@
 
 #include "cluster/topic_table_probe.h"
 
+#include "cluster/cluster_utils.h"
 #include "cluster/topic_table.h"
 #include "config/configuration.h"
+#include "config/node_config.h"
 #include "prometheus/prometheus_sanitize.h"
 
 namespace cluster {
 
 topic_table_probe::topic_table_probe(const topic_table& topic_table)
-  : _topic_table(topic_table) {}
+  : _topic_table(topic_table) {
+    setup_metrics();
+}
+
+void topic_table_probe::setup_metrics() {
+    setup_public_metrics();
+    setup_internal_metrics();
```

Comment:
I don't like this, because if we later need to add another metric to the public, and not to the internal one, we will need to return the function back.